### PR TITLE
Check if osd hosts support osd_mkfs_type filesystem

### PR DESCRIPTION
**Pull request by harobed**
Check if osd hosts support osd_mkfs_type filesystem #1406

### CODE REVIEW

**Comment by harobed** [comment]
`xfs` isn't supported by Ubuntu "xenial" ?

What is purge cluster ?

File: roles/ceph-osd/tasks/check_system.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+- name: fail on unsupported {{osd_mkfs_type}} filesystem (configured by osd_mkfs_type variable)
+  command: grep -q "[[:space:]]{{osd_mkfs_type}}$" /proc/filesystems
```
Comment (leseb) [inline roles/ceph-osd/tasks/check_system.yml]: Please add to the task:

```
changed_when: false
failed_when: false
```

Can you also add a `fail` task to fail properly with a useful message?

Comment (harobed) [inline roles/ceph-osd/tasks/check_system.yml]: Ok, I'll do that.

Comment (leseb) [inline roles/ceph-osd/tasks/check_system.yml]: Can you do this ASAP? We want to tag stable-2.2 by the end of the week.
Thanks!

**Comment by zhsj** [comment]
Hi, the xfs module is not loaded by default on Debian, but it has. So the check for /proc/filesystem is not accurate.

/proc/filesystem only shows the fs module used currently. But some fs kernel modules not loaded are also available to use.

**Comment by leseb** [comment]
Closing this due to non-activity, feel free to re-open.

**Pull request closed without merging**